x-pack/plugins/cases/public/components/case_view/components/case_view_activity.tsx

```diff
@@ -229,17 +226,20 @@ export const CaseViewActivity = ({
             onSubmit={onSubmitTags}
             isLoading={isLoading && loadingKey === 'tags'}
           />
-          {pushToServiceAuthorized && userActionsData ? (
+          {pushToServiceAuthorized &&
+          userActionsData &&
+          caseConnectors &&
+          caseData &&
+          allAvailableConnectors ? (
             <EditConnector
               caseData={caseData}
-              caseServices={userActionsData.caseServices}
-              connectorName={connectorName}
-              connectors={connectors}
-              hasDataToPush={userActionsData.hasDataToPush}
-              isLoading={isLoadingConnectors || (isLoading && loadingKey === 'connector')}
-              isValidConnector={isLoadingConnectors ? true : isValidConnector}
+              caseConnectors={caseConnectors}
+              allAvailableConnectors={allAvailableConnectors}
+              isLoading={
+                isLoadingAllAvailableConnectors || (isLoading && loadingKey === 'connector')
+              }
               onSubmit={onSubmitConnector}
-              userActions={userActionsData.caseUserActions}
+              key={caseData.connector.id}
```

Comment:
We need to reset the state of the `EditConnector` component every time the current connector changes. We use the `key` do to that. This helps to remove all unnecessary `useEffect`s that update the state when the `caseData` changes.